hedera-node/hedera-network-admin-service-impl/src/test/java/com/hedera/node/app/service/networkadmin/impl/test/handlers/NetworkGetByKeyHandlerTest.java

```diff
@@ -0,0 +1,85 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.networkadmin.impl.test.handlers;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hedera.node.app.spi.fixtures.workflows.ExceptionConditions.responseCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.hedera.hapi.node.base.QueryHeader;
+import com.hedera.hapi.node.base.ResponseHeader;
+import com.hedera.hapi.node.transaction.GetByKeyQuery;
+import com.hedera.hapi.node.transaction.GetByKeyResponse;
+import com.hedera.hapi.node.transaction.Query;
+import com.hedera.hapi.node.transaction.Response;
+import com.hedera.node.app.service.networkadmin.impl.handlers.NetworkGetByKeyHandler;
+import com.hedera.node.app.spi.workflows.PreCheckException;
+import com.hedera.node.app.spi.workflows.QueryContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class NetworkGetByKeyHandlerTest {
+    @Mock
+    private QueryContext context;
+
+    private NetworkGetByKeyHandler subject;
+
+    @BeforeEach
+    void setUp() {
+        subject = new NetworkGetByKeyHandler();
+    }
+
+    @Test
+    void extractsHeader() {
+        final var data = GetByKeyQuery.newBuilder()
+                .header(QueryHeader.newBuilder().build())
+                .build();
+        final var query = Query.newBuilder().getByKey(data).build();
+        final var header = subject.extractHeader(query);
+        final var op = query.getByKeyOrThrow();
+        assertEquals(op.header(), header);
+    }
+
+    @Test
+    void createsEmptyResponse() {
+        final var responseHeader = ResponseHeader.newBuilder().build();
+        final var response = subject.createEmptyResponse(responseHeader);
+        final var expectedResponse = Response.newBuilder()
+                .getByKey(GetByKeyResponse.newBuilder().header(responseHeader))
+                .build();
+        assertEquals(expectedResponse, response);
+    }
+
+    @Test
+    void validateThrowsPreCheck() {
+        assertThatThrownBy(() -> subject.validate(context))
+                .isInstanceOf(PreCheckException.class)
+                .has(responseCode(NOT_SUPPORTED));
+    }
+
+    @Test
+    void findResponseThrowsUnsupported() {
+        final var responseHeader = ResponseHeader.newBuilder().build();
+        assertThrows(UnsupportedOperationException.class, () -> subject.findResponse(context, responseHeader));
```

Comment:
Again, this is the JUnit `assertThrows`, whereas the AssertJ `assertThatThrownBy` is used above.

Reply:
Fixed